cartridges/plugin_beam/cartridge/templates/default/beam/beam_cumulative_impact.isml

```diff
@@ -1,9 +1,21 @@
 <isset value="${dw.system.Site.current.preferences.custom}" name="sitePref" scope="page"/>
-<isset value="${request.getHttpLocale()}" name="currentLocale" scope="page"/>
-<isset value="${currentLocale.split('_')[0]}" name="languageCode" scope="page"/>
-<isset value="${sitePref.beamSupportedLang}" name="supportedLang" scope="page" />
-<isset value="${supportedLang && supportedLang.split(',').indexOf(languageCode) !== -1 ? languageCode : sitePref.beamDefaultLang}" name="finalLanguageCode" scope="page"/>
 
+<isscript>
+    var Site = require('dw/system/Site');
+    var Locale = require('dw/util/Locale');
+    var languageHelper = require('*/cartridge/scripts/utils/getFinalLanguageCode'); // Import the helper function
+
+    var currentLanguageCode = Locale.getLocale(request.locale).getLanguage();
+
+    var finalLanguageCode = languageHelper.getFinalLanguageCode(Site.getCurrent(), currentLanguageCode);
+
+    session.privacy.finalLanguageCode = finalLanguageCode;
+
+    require('dw/system/Logger').debug("Current Language Code: {0}", currentLanguageCode);
+    require('dw/system/Logger').debug("Final Language Code: {0}", finalLanguageCode);
```

Comment:
`request` Sorry, missed these in my prior review. If we don't need these Loggers, we should probably remove them as well.